kotlin-checks-test-sources/src/main/kotlin/checks/PropertyGetterAndSetterUsageCheckSample.kt

```diff
@@ -19,14 +21,172 @@ open class PropertyGetterAndSetterUsageCheckSample {
         fun getName(): String = name // Noncompliant {{Convert this getter to a "get()" on the property "name".}}
 //          ^^^^^^^
     }
-    private class PrivateClassOnlySetter {
-        private var id: String = ""
-        //          ^^> {{Property "id"}}
-        fun setId(id: String) { // Noncompliant {{Convert this setter to a "set(value)" on the property "id".}}
-//          ^^^^^
-            this.id = id
+
+    open class GetterAndSetterVisibilityCompatibility {
+        // Kotlin's properties only support setter visibility <= getter visibility
+        // So the rule should not raise issue when accessors visibility is not compatible with Kotlin's properties
+
+        // ______________________________________________
+        // 1) Setter with visibility <= getter visibility
+
+        // public getter / public setter
+        private var f1: Int = 0
+        fun getF1(): Int = f1 // Noncompliant
+        fun setF1(value: Int) { f1 = value }
+
+        // public getter / internal setter
+        private var f2: Int = 0
+        fun getF2(): Int = f2 // Noncompliant
+        internal fun setF2(value: Int) { f2 = value }
+
+        // public getter / protected setter
+        private var f3: Int = 0
+        fun getF3(): Int = f3 // Noncompliant
+        protected fun setF3(value: Int) { f3 = value }
+
+        // public getter / private setter
+        private var f4: Int = 0
+        fun getF4(): Int = f4 // Noncompliant
+        private fun setF4(value: Int) { f4 = value }
+
+        // internal getter / internal setter
+        private var f5: Int = 0
+        internal fun getF5(): Int = f5 // Noncompliant
+        internal fun setF5(value: Int) { f5 = value }
+
+        // internal getter / private setter
+        private var f6: Int = 0
+        internal fun getF6(): Int = f6 // Noncompliant
+        private fun setF6(value: Int) { f6 = value }
+
+        // protected getter / protected setter
+        private var f7: Int = 0
+        protected fun getF7(): Int = f7 // Noncompliant
+        protected fun setF7(value: Int) { f7 = value }
+
+        // protected getter / private setter
+        private var f8: Int = 0
+        protected fun getF8(): Int = f8 // Noncompliant
+        private fun setF8(value: Int) { f8 = value }
+
+        // private getter / private setter
+        private var f9: Int = 0
+        private fun getF9(): Int = f9 // Noncompliant
+        private fun setF9(value: Int) { f9 = value }
+
+        // ______________________________________________
+        // 2) Setter with visibility > getter visibility
```

Comment:
I would still expect the getter in these examples to use the property access syntax here, so for the `f10` example, to look like something like this:
```
        internal var f10: Int = 0
        public fun setF10(value: Int) { f10 = value }
```

Reply:
It's not equivalant, your proposal expose an implicit `internal set(value) { field = value }` property setter. Having two setters could be dangerous if there's a specific logic that must be called in `setF10`. The smallest equivalent replacement could be:

- Before

```kotlin
    private var f10: Int = 0
    internal fun getF10(): Int = /* ... getter logic */ f10
    public fun setF10(value: Int) { /* ... setter logic */ f10 = value }
```

- After

```kotlin
    internal var f10: Int = 0
        get() = /* getter logic */ field
        private set // hide the implicit setter
    public fun setF10(value: Int) { /* ... setter logic */ f10 = value }
```

So for now my fear is to ask a developer to change the code without enough assistance about the implicit setter visibility, and this could become a bug. I prefer to not raise an issue with the current sate of the rule description, and when we will work again to improve this rule (see [SONARKT-308](https://sonarsource.atlassian.net/jira/software/c/projects/SONARKT/issues/SONARKT-308)), we could improve the educational section about implicit getter and setter.